### PR TITLE
ilbgateway service Annotations not only for gke

### PR DESCRIPTION
**Pull request by jackge007**

### CODE REVIEW

**Comment by jackge007** [comment]
I singed !

File: install/kubernetes/helm/istio/charts/gateways/values.yaml [hunk 1]
```diff
@@ -253,8 +253,11 @@ istio-ilbgateway:
     #  cpu: 1800m
     #  memory: 256Mi
   loadBalancerIP: ""
-  serviceAnnotations:
-    cloud.google.com/load-balancer-type: "internal"
+  serviceAnnotations: {}
+  #  cloud.google.com/load-balancer-type: "internal"
```
Comment (sdake) [inline install/kubernetes/helm/istio/charts/gateways/values.yaml]: there is a small problem with this change.  no `serviceAnnotations` annotation is defined by default.

**Comment by sdake** [comment]
> It isn't clear to me what you are trying to do here. It looks to me like you just moved the values around, but the actual values are the same (as in, there is no real change to the generated yaml here). Am I missing something?

@howardjohn what was done was that the annotations for other cloud providers (aws, azure), were added as comments to `values.yaml`.  This PR can't merge as is however, as it has no default for ilb annotations.

File: install/kubernetes/helm/istio/values.yaml [hunk 2]
```diff
@@ -13,7 +13,13 @@
 #
 gateways:
   enabled: true
-
+  istio-ilbgateway: 
+    enabled: false
+    serviceAnnotations:
+      cloud.google.com/load-balancer-type: "internal"
+      #  service.beta.kubernetes.io/alicloud-loadbalancer-address-type: "intranet"
```
Comment (howardjohn) [inline install/kubernetes/helm/istio/values.yaml]: I am not sure we want these in the top level values file. We try to keep this one small (just `enabled`) and have the settings come from the chart (gateways/values.yaml)

File: install/kubernetes/helm/istio/charts/gateways/values.yaml (hunk 1 above)
Comment (howardjohn) [reply install/kubernetes/helm/istio/charts/gateways/values.yaml]: +1, adding the comments is ok with me as long as we keep the default

**Pull request closed without merging**